common/djangoapps/contentstore_shared/views/tabs.py

```diff
@@ -0,0 +1,206 @@
+        tab = get_tab_by_tab_id_locator(old_tab_list, tab_id_locator)
+        if tab is None:
+            return JsonResponse(
+                {"error": "Tab with id_locator '{0}' does not exist.".format(tab_id_locator)}, status=400
+            )
+        new_tab_list.append(tab)
+
+    # the old_tab_list may contain additional tabs that were not rendered in the UI because of
+    # global or course settings.  so add those to the end of the list.
+    non_displayed_tabs = set(old_tab_list) - set(new_tab_list)
+    new_tab_list.extend(non_displayed_tabs)
+
+    # validate the tabs to make sure everything is Ok (e.g., did the client try to reorder unmovable tabs?)
+    try:
+        CourseTabList.validate_tabs(new_tab_list)
+    except InvalidTabsException, exception:
+        return JsonResponse(
+            {"error": "New list of tabs is not valid: {0}.".format(str(exception))}, status=400
+        )
+
+    # persist the new order of the tabs
+    course_item.tabs = new_tab_list
+    modulestore().update_item(course_item, request.user.id)
+
+    return JsonResponse()
+
+
+def edit_tab_handler(course_item, request):
+    """
+    Helper function for handling requests to edit settings of a single tab
+    """
+
+    # Tabs are identified by tab_id or locator
+    tab_id_locator = request.json['tab_id_locator']
+
+    # Find the given tab in the course
+    tab = get_tab_by_tab_id_locator(course_item.tabs, tab_id_locator)
+    if tab is None:
+        return JsonResponse(
+            {"error": "Tab with id_locator '{0}' does not exist.".format(tab_id_locator)}, status=400
+        )
+
+    if 'is_hidden' in request.json:
+        # set the is_hidden attribute on the requested tab
+        tab.is_hidden = request.json['is_hidden']
+        modulestore().update_item(course_item, request.user.id)
+    else:
+        raise NotImplementedError('Unsupported request to edit tab: {0}'.format(request.json))
+
+    return JsonResponse()
+
+
+def get_tab_by_tab_id_locator(tab_list, tab_id_locator):
+    """
+    Look for a tab with the specified tab_id or locator.  Returns the first matching tab.
+    """
+    if 'tab_id' in tab_id_locator:
+        tab = CourseTabList.get_tab_by_id(tab_list, tab_id_locator['tab_id'])
+    elif 'tab_locator' in tab_id_locator:
+        tab = get_tab_by_locator(tab_list, tab_id_locator['tab_locator'])
+    return tab
+
+
+def get_tab_by_locator(tab_list, usage_key_string):
+    """
+    Look for a tab with the specified locator.  Returns the first matching tab.
+    """
+    tab_location = UsageKey.from_string(usage_key_string)
+    item = modulestore().get_item(tab_location)
+    static_tab = StaticTab(
+        name=item.display_name,
+        url_slug=item.location.name,
+    )
+    return CourseTabList.get_tab_by_id(tab_list, static_tab.tab_id)
+
+
+# "primitive" tab edit functions driven by the command line.
+# These should be replaced/deleted by a more capable GUI someday.
```

Comment:
As discussed F2F:
- remove the `content_shared` djangoapp from `common`
- move the `primitive_*` functions (and `validate_args`) into
  common, maybe in `xmodule.tabs`
- update the (existing) CMS management command
  (`cms/djangoapps/contentstore/management/commands/edit_course_tabs.py`)
  to reference the new import path (`xmodule.tabs.*`)
- update your new sysadmin code to reference the new import path
  (`xmodule.tabs.*)
- remove the TODO note :) [1] CC: @nparlante

[1] https://github.com/Stanford-Online/edx-platform/pull/234/files#diff-6e3ab72cd3e6611314ddb9c85939a48eR177